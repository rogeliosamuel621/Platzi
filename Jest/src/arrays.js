const fruits = ['banana', 'orange', 'apple', 'lemon'];

const colors = ['blue', 'green', 'red', 'pink', 'yellow'];

export const arraysFruits = () => fruits;
export const arraysColors = () => colors;
