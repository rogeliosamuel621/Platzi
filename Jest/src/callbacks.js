export function callbackHell(callback) {
  callback('hola javascripters');
}
