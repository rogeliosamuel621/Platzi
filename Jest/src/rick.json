{
  "id": 1,
  "name": "Rick sanches",
  "status": "Alive",
  "specie": "Human",
  "gender": "Male"
}