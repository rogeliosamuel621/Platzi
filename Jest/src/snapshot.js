export const getCharacter = (data) => {
  return {
    id: data.id,
    name: data.name,
    status: data.status,
    species: data.species,
    gender: data.gender,
  };
};
