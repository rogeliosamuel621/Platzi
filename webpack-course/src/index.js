import sum from './utils/sum';

const hello = 'Hello';

console.log(hello);
console.log(sum(1, 2));
