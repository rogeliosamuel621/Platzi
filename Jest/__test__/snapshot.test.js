import { getCharacter } from '../src/snapshot';
import rick from '../src/rick.json';

describe('Snapshots', () => {
  test('Snapshot test', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test('always fail', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });

  test('have an exception', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Rogelio',
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
