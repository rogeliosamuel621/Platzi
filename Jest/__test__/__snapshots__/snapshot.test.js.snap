// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Snapshot test 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick sanches",
  "species": undefined,
  "status": "Alive",
}
`;

exports[`Snapshots always fail 1`] = `
Object {
  "createdAt": 2020-11-01T13:58:38.810Z,
  "id": 17,
}
`;

exports[`Snapshots have an exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Rogelio",
}
`;
